src/pipeline.ml

```diff
@@ -57,223 +57,250 @@ let include_git = { Cluster_api.Docker.Spec.defaults with include_git = true }
+        | `PR pri -> `PR pri.Github.Api.Ref.id
+        in
+        Current.map Github.Api.Commit.id @@ Github.Api.head_of api repo id_type
+      | None -> Github.Api.Anonymous.head_of repo id
+    in
+    let additional_build_args =
+      let+ opam_repository_commit =
+        head_of { Github.Repo_id.owner = "ocaml"; name = "opam-repository" } @@ `Ref "refs/heads/master"
+      and+ platform_blog_commit =
+        head_of { Github.Repo_id.owner = "ocaml"; name = "platform-blog" } @@ `Ref "refs/heads/master"
+      and+ opam_commit =
+        head_of { Github.Repo_id.owner = "ocaml"; name = "opam" } @@ `Ref "refs/heads/master"
       in
-      Current.map Github.Api.Commit.id @@ Github.Api.head_of api repo id_type
-    | None -> Github.Api.Anonymous.head_of repo id
-  in
+      [
+        "OPAM_REPO_GIT_SHA=" ^ Current_git.Commit_id.hash opam_repository_commit;
+        "BLOG_GIT_SHA=" ^ Current_git.Commit_id.hash platform_blog_commit;
+        "OPAM_GIT_SHA=" ^ Current_git.Commit_id.hash opam_commit
+      ]
+    in
+    pipelines, additional_build_args
 
-  let additional_build_args =
-    let+ opam_repository_commit =
-      head_of { Github.Repo_id.owner = "ocaml"; name = "opam-repository" } @@ `Ref "refs/heads/master"
-    and+ platform_blog_commit =
-      head_of { Github.Repo_id.owner = "ocaml"; name = "platform-blog" } @@ `Ref "refs/heads/master"
-    and+ opam_commit =
-      head_of { Github.Repo_id.owner = "ocaml"; name = "opam" } @@ `Ref "refs/heads/master" in
-    ["OPAM_REPO_GIT_SHA=" ^ Current_git.Commit_id.hash opam_repository_commit;
-     "BLOG_GIT_SHA=" ^ Current_git.Commit_id.hash platform_blog_commit;
-     "OPAM_GIT_SHA=" ^ Current_git.Commit_id.hash opam_commit]
-  in
+  let extras () =
+    let tarsnap =
+      let monthly = Current_cache.Schedule.v ~valid_for:(Duration.of_day 30) () in
+      Current_ssh.run ~schedule:monthly "watch.ocaml.org" ~key:"tarsnap" (Current.return ["./tarsnap-backup.sh"])
+    in
+    let peertube =
+      let weekly = Current_cache.Schedule.v ~valid_for:(Duration.of_day 7) () in
+      let image = Cluster.Watch_docker.pull ~schedule:weekly "chocobozzz/peertube:production-bookworm" in
+      Cluster.Watch_docker.service ~name:"infra_peertube" ~image ()
+    in
+    [tarsnap; peertube]
 
-  let opam_repository_pipeline = filter_list filter [
-    ocaml_opam, "opam2web", [
-      docker_registry (Duration.of_min 360) "Dockerfile"
-      ["live", "opam.ocaml.org:live", [`Ocamlorg_opam4 "infra_opam_live"; `Ocamlorg_opam5 "infra_opam_live"];
-       "live-staging", "opam.ocaml.org:staging", [`Ocamlorg_opam4 "infra_opam_staging"; `Ocamlorg_opam5 "infra_opam_staging"]]
-    ];
-  ]
-  in
+  let v ?app ?notify:channel ?filter ~sched ~staging_auth () =
+    (* [web_ui collapse_value] is a URL back to the deployment service, for links
+      in status messages. *)
+    let web_ui repo = Uri.with_query' base_url ["repo", repo] in
+    let build ?additional_build_args (org, name, builds) =
```

Comment:
Unless I am mistaken, it doesn't look like the optional argument `?additional_build_args` ever gets used for this `build` function. In which case I guess we wouldn't need it?

Reply:
I'm not sure I understand, the additional build args are `OPAM_REPO_GIT_SHA`, `BLOG_GIT_SHA`, and `OPAM_GIT_SHA` and are passed to the cluster builder.

Reply:
Unless I am misreading, this line defines a new helper function `build` which takes an optional argument. That optional argument is not used at the one place the helper function is invoked (line 254). So the optional argument here isn't used, and (since it is optional) we can just call `Cluster_build.repo` omitting the `?additional_build_args`.

This is in contrast to `build_for_registry` where the optional argument is in fact used.

Reply:
Yes you're right, I missed that, apologies for pulling the trigger on merging. I'll include this in the next PR.